finagle-core/src/main/scala/com/twitter/finagle/filter/RequestMeterFilter.scala

```diff
@@ -0,0 +1,21 @@
+package com.twitter.finagle.filter
+
+import com.twitter.concurrent.AsyncMeter
+import com.twitter.finagle.{Failure, Service, SimpleFilter}
+import com.twitter.util.{Future, Throw}
+
+/**
+  * A [[com.twitter.finagle.Filter]] that rate limits requests to a fixed rate over time by
+  * using the [[com.twitter.concurrent.AsyncMeter]] implementation. It can be used for slowing
+  * down access to throttled resources. Requests that are unable to acquire a permit are failed
+  * immediately with a [[com.twitter.finagle.Failure]] that signals a restartable or
+  * idempotent process.
```

Comment:
suppose you're talking to the foursquare API.  they can handle more traffic than you're throwing at them, but as a policy, they have a ratelimit of N / second.  so you want to rate limit in order to protect them–if you were talking to a service that you owned, it could be that you want to send them traffic as fast as they can handle it, but within reason, ie not more than fifty concurrent requests.  in the foursquare example, they're artificially throttling–in the example of your service, it will be able to speed up or slow down as your service is fast or slow.